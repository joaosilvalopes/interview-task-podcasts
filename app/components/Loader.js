import styled from "styled-components";

const RING_SIZE = 3.1;
const CIRCLE_SIZE = 1.5;

const Container = styled.div`
    position: relative;
    width: ${RING_SIZE}rem;
    height: ${RING_SIZE}rem;
`;

const Circle = styled.div`
    position: absolute;
    top: ${RING_SIZE / 2 - CIRCLE_SIZE / 2}rem;
    left: ${RING_SIZE / 2 - CIRCLE_SIZE / 2}rem;
    width: ${CIRCLE_SIZE}rem;
    height: ${CIRCLE_SIZE}rem;
    border-radius: 50%;
    background-color: blue;
`;

const Ring = styled.div`
    @keyframes pulsate {
        0% {
            transform: scale(0.1, 0.1);
            opacity: 0;
        }
        50% {
            opacity: 1;
        }
        100% {
            transform: scale(1.2, 1.2);
            opacity: 0;
        }
    }

    position: absolute;
    top: 0;
    left: 0;
    height: ${RING_SIZE}rem;
    width: ${RING_SIZE}rem;
    border: 0.3rem solid blue;
    border-radius: 50%;
    animation: pulsate 1s ease-out infinite; 
    opacity: 0;
`;

const Loader = (props) => (
    <Container {...props}>
        <Ring></Ring>
        <Circle></Circle>
    </Container>
)

export default Loader;
