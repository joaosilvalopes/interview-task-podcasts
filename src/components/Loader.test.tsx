import { render } from '@testing-library/react';

import Loader from './Loader';

it('should render correctly', async () => {
    const { asFragment } = render(<Loader />);
  
    expect(asFragment()).toMatchSnapshot();
});
