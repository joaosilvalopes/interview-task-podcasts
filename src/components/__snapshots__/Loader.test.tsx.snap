// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe eglgry"
  >
    <div
      class="sc-dkrFOg eZaBCl"
    />
    <div
      class="sc-gswNZR eMufKk"
    />
  </div>
</DocumentFragment>
`;
